pkg/api/defaults-scheduler.go

```diff
@@ -36,4 +38,11 @@ func (cs *ContainerService) setSchedulerConfig() {
 	for key, val := range staticSchedulerConfig {
 		o.KubernetesConfig.SchedulerConfig[key] = val
 	}
+
+	invalidFeatureGates := []string{}
+	// Remove --feature-gate VolumeSnapshotDataSource starting with 1.22
+	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.22.0-alpha.1") {
+		invalidFeatureGates = append(invalidFeatureGates, "VolumeSnapshotDataSource")
```

Comment:
is the `VolumeSnapshotDataSource` really present on all these components? My understanding was that it was only added to the apiserver

Reply:
Based on previous k8s version v1.19 documentation, `VolumeSnapshotDataSource` was a valid feature gate to specify in all these components.